cpp/src/arrow/array/statistics.h

```diff
@@ -34,6 +35,22 @@ namespace arrow {
 struct ARROW_EXPORT ArrayStatistics {
   using ValueType = std::variant<bool, int64_t, uint64_t, double, std::string>;
 
+  static const std::shared_ptr<DataType>& ValueToArrowType(
+      const std::optional<ValueType>& value) {
+    if (!value.has_value()) {
+      return null();
+    }
+
+    struct Visitor {
+      const std::shared_ptr<DataType>& operator()(const bool&) { return boolean(); }
+      const std::shared_ptr<DataType>& operator()(const int64_t&) { return int64(); }
+      const std::shared_ptr<DataType>& operator()(const uint64_t&) { return uint64(); }
+      const std::shared_ptr<DataType>& operator()(const double&) { return float64(); }
+      const std::shared_ptr<DataType>& operator()(const std::string&) { return utf8(); }
```

Comment:
I may forgot a bit but we don't distinct "bytes" and "utf8" in stats?

Reply:
Ah, we didn't discuss it...
Let's discuss it in #44579.

We can assume "utf8" here for now.

Reply:
We can add a `// TODO(GH-44579)` here?

Reply:
Yes. I should have added it...
I've added it.

Reply:
Oh, I forgot to push the commit... I pushed now.